velox/expression/CastExpr-inl.h

```diff
@@ -380,7 +381,17 @@ void CastExpr::applyCastKernel(
         ToKind == TypeKind::VARCHAR || ToKind == TypeKind::VARBINARY) {
       // Write the result output to the output vector
       auto writer = exec::StringWriter<>(result, row);
-      writer.copy_from(output);
+      // For target VARCHAR cast take type length into account.
```

Comment:
Let's make sure to document semantics of cast to/from the new type: https://facebookincubator.github.io/velox/functions/presto/conversion.html